src/main/java/ch/njol/skript/events/EvtBucketEntity.java

```diff
@@ -0,0 +1,103 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.util.Getter;
+import ch.njol.util.StringUtils;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerBucketEntityEvent;
+import org.bukkit.event.player.PlayerFishEvent;
+import org.bukkit.inventory.ItemStack;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Name("Bucket Capture Entity")
+@Description("Called when a player captures an entity in a bucket.")
+@Examples({
+	"on bucket capture of a puffer fish:",
+		"\tsend \"You caught a fish with your bucket!\" to player"
+})
+@RequiredPlugins("MC 1.17+")
+@Since("INSERT VERSION")
+public class EvtBucketEntity extends SkriptEvent {
+
+	static {
+		if (Skript.classExists("org.bukkit.event.player.PlayerBucketEntityEvent")) {
+			Skript.registerEvent("Bucket Capture Entity", EvtBucketEntity.class, PlayerFishEvent.class, "[player] bucket captur(e|ing) [[of] %-entitydatas%]");
+
+			EventValues.registerEventValue(PlayerBucketEntityEvent.class, ItemStack.class, new Getter<ItemStack, PlayerBucketEntityEvent>() {
+				@Override
+				public ItemStack get(PlayerBucketEntityEvent event) {
+					return event.getOriginalBucket();
+				}
+			}, EventValues.TIME_NOW);
+			EventValues.registerEventValue(PlayerBucketEntityEvent.class, Player.class, new Getter<Player, PlayerBucketEntityEvent>() {
+				@Override
+				public Player get(PlayerBucketEntityEvent event) {
+					return event.getPlayer();
+				}
+			}, EventValues.TIME_NOW);
+			EventValues.registerEventValue(PlayerBucketEntityEvent.class, Entity.class, new Getter<Entity, PlayerBucketEntityEvent>() {
+				@Override
+				@Nullable
+				public Entity get(PlayerBucketEntityEvent event) {
+					return event.getEntity();
+				}
+			}, EventValues.TIME_NOW);
+		}
+	}
+
+	private List<EntityData<?>> entities = new ArrayList<>(0);
```

Comment:
This could be an array. The value is only ever being set once, a list is for dynamically adding multiple values.
```suggestion
	private EntityData<?>[] entities = null;
```
Requires updating other parts of code.